### ISSUE TITLE
Incorrect union type report in try catch

### ISSUE THREAD

**Issue by kaliberx** [PHPCSStandards/PHP_CodeSniffer#660]
```
$ cat error.php
<?php

try {
    // nothing
} catch (Exception|RuntimeException) {
}

 ./phpcs.phar --standard=PSR12 error.php

FILE: .../error.php
---------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
---------------------------------------------------------------------------
 5 | ERROR | [x] Expected at least 1 space before "|"; 0 found
 5 | ERROR | [x] Expected at least 1 space after "|"; 0 found
---------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
---------------------------------------------------------------------------
```

## Expected behavior

No error reported.

## Please confirm
Sorry, tested on 3.10.3, not on the master.

**Comment by jrfnl** [comment]
@kaliberx And that's exactly as it should be per: https://www.php-fig.org/psr/psr-12/#56-try-catch-finally

Keep in mind that when PSR12 was written, the concept of "union types"in PHP didn't exist yet. The expression in a catch condition is seen as a bitmask with a bitwise `or` operator, not a union type.

**Comment by klausi** [comment]
The problem is that PER 3.0 wants to have it without spaces in catch(): https://www.php-fig.org/per/coding-style/#56-try-catch-finally

Do we need to start new PER 3.0 sniffs in PHPCS?

**Comment by jrfnl** [comment]
@klausi I don't think so. I think we need a property which is (re-)used by multiple sniffs as described here: https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/566#issuecomment-2358618753

**Comment by klausi** [comment]
Created an attempt in #1356 , let me know if you need any changes!

**Comment by rodrigoprimo** [comment]
A similar report was previously discussed in the original repository: squizlabs/PHP_CodeSniffer#3663. Linking for visibility.
